File: Map/Map/Map/statement/IfStatement.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Map.domain;
using Map.expression;

namespace Map.statement
{
    class IfStatement : IStatement
    {
        private IExpression cond;
        private IStatement trueSt, falseSt;

        public IfStatement(IExpression e, IStatement trueS, IStatement falseS)
        {
            cond = e;
            trueSt = trueS;
            falseSt = falseS;
        }

        public ProgramState execute(ProgramState ps)
        {
            if (cond.eval(ps.symbolTable) == 0)
                ps.exeStack.push(falseSt);
            else
                ps.exeStack.push(trueSt);
            return null;
        }

        public override String ToString()
        {
            return "if " + cond + " then " + trueSt + " else " + falseSt;
        }
    }
}

File: Map/Z_InterpretorWPro/src/statement/IfStatement.java
package statement;

import domain.IDictionary;
import domain.IHeap;
import domain.IStack;
import domain.PrgState;
import expression.IExpression;

public class IfStatement implements IStatement
{
	private IExpression condition;
	private IStatement trueStatement,falseStatement;
	
	public IfStatement(IExpression e, IStatement trueS, IStatement falseS)
	{
		condition = e;
		trueStatement = trueS;
		falseStatement = falseS;
	}

	@Override
	public PrgState execute(PrgState ps) 
	{
		IStack<IStatement> stack = ps.getExeStack();
		IDictionary<String,Integer> dict = ps.getStackOfSym().first();
		IHeap<Integer,Integer> heap = ps.getHeap();
		if(condition.eval(dict,heap)==0)
		{
			stack.push(falseStatement);
		}
		else
		{
			stack.push(trueStatement);
		}
		return null;
	}
	
	@Override
	public String toString()
	{
		return "if "+condition+" then "+trueStatement+" else "+falseStatement;
	}
	
	
	
	
}
